src/input.cpp

```diff
@@ -449,7 +451,8 @@ static void input_mapping_execute(const input_mapping_t &m, bool allow_commands)
 
     // !has_functions && !has_commands: only set bind mode
     if (!has_commands && !has_functions) {
-        input_set_bind_mode(m.sets_mode);
+        if (! m.sets_mode.empty())
+            input_set_bind_mode(m.sets_mode);
```

Comment:
This should either be on the same line as the `if()` or enclosed in braces. I've actually seen this pattern (a single line block not enclosed in braces) lead to bugs. And no space after the `!` (`make style` will just remove it). Also the block at line 495.